cmd/encryption-v1.go

```diff
@@ -386,34 +262,101 @@ func newEncryptReader(content io.Reader, key []byte, bucket, object string, meta
 	return reader, nil
 }
 
+// generateKMSKey gets a new random plaintext key and sealed plaintext key from
+// Vault. It returns the plaintext key and stores the master Key ID and
+// the sealed plaintext key in the metadata map on success.
+func generateKMSKey(r *http.Request, bucket, object string, metadata map[string]string) ([]byte, error) {
+	if globalKMS == nil {
+		return nil, errKMSNotConfigured
+	}
+	if crypto.S3.IsRequested(r.Header) && crypto.SSEC.IsRequested(r.Header) {
+		return nil, crypto.ErrIncompatibleEncryptionMethod
+	}
+	ctx := crypto.Context{bucket: path.Join(bucket, object)}
+	key, encKey, err := globalKMS.GenerateKey(globalKMSKeyID, ctx)
+	if err != nil {
+		return nil, err
+	}
+	metadata[crypto.S3KMSKeyID] = globalKMSKeyID
+	metadata[crypto.S3KMSSealedKey] = string(encKey)
+	return key[:], nil
+}
+
+// create new encryption metadata from http request headers for SSE-C and generated key from KMS in the case of
+// SSE-S3
+func extractEncryptionMetadata(r *http.Request, bucket, object string, metadata map[string]string) (err error) {
```

Comment:
This function looks more like a validation, not extracting and returning values, can this function be renamed to a better name ?

Reply:
@vadmeste most of the code will go away anyway as soon as I refactor SSE code completely. However suggestion is okay IMO...

Reply:
renamed this.